### PR TITLE
[MUI-535]: Text input component

### PR DESCRIPTION
**Pull request by mr-teddy-bear**
# Description

Text input component

## Jira ticket(optional)

Attach link to Jira [ticket]. 

## Type of change

Please pick options that are relevant.

- [ ] Bug fix OR refactoring (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature or refactoring that would cause existing functionality to not work as expected)

### CODE REVIEW

File: src/components/Input/Input.tsx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+export const Input = ({
+  showError,
+  errorText,
+  success,
+  labelText,
+  ...inputProps
+}: InputBaseProps & InputCustomProps): JSX.Element => {
+  return (
+    <FormControl fullWidth error={showError} variant='standard'>
+      <InputLabel shrink htmlFor={inputProps.id}>
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:62]: label not meet figma

File: src/components/Input/Input.tsx [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        {labelText}
+      </InputLabel>
+      <BaseInput className={success ? 'Mui-success' : ''} {...inputProps} />
+      <Typography
+        variant='subtitle2'
+        text={showError ? String(errorText) : ''}
+        style={{ color: '#DC1115', marginTop: '4px', height: '16px' }}
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:69]: user sx instead of style

File: src/components/Input/Input.tsx [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  success,
+  labelText,
+  ...inputProps
+}: InputBaseProps & InputCustomProps): JSX.Element => {
+  return (
+    <FormControl fullWidth error={showError} variant='standard'>
+      <InputLabel shrink htmlFor={inputProps.id}>
+        {labelText}
+      </InputLabel>
+      <BaseInput className={success ? 'Mui-success' : ''} {...inputProps} />
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:65]: className can be overwriten by inputProps

File: src/components/Input/Input.tsx [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  labelText?: string;
+};
+
+const BaseInput = styled(InputBase)(({ theme }) => ({
+  'label + &': {
+    marginTop: theme.spacing(3),
+  },
+  '& .MuiInputBase-input': {
+    borderRadius: 4,
+    position: 'relative',
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:23]: for what here position relative?

File: src/components/Input/Input.tsx [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+
+import FormControl from '@mui/material/FormControl';
+import InputBase, { InputBaseProps } from '@mui/material/InputBase';
+import InputLabel from '@mui/material/InputLabel';
+import { styled } from '@mui/material/styles';
+
+import Typography from '../Typography/Typography';
+
+type InputCustomProps = {
+  showError?: boolean;
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:11]: I think we can leave only error props. If error text exist that's mean error present. What do you think?

File: src/components/Input/Input.tsx [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+  errorText?: string;
+  success?: boolean;
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:13]: Sorry, I can't find success state in figma. Can you share it please?

**Comment by DmytroMinterest** [comment]
Check Example mapping of this issue please: [MUI-535](https://minterestfinance.atlassian.net/browse/MUI-535)

File: src/components/Input/Input.tsx (hunk 1 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx:62]: Done

File: src/components/Input/Input.tsx (hunk 2 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx:69]: Done

File: src/components/Input/Input.tsx (hunk 3 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx:65]: Done

File: src/components/Input/Input.tsx (hunk 4 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx:23]: Done

File: src/components/Input/Input.tsx (hunk 6 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx:13]: Done

File: src/components/Input/Input.tsx [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+  return (
+    <FormControl fullWidth error={!!errorText} variant='standard'>
+      <Label htmlFor={inputProps.id}>
+        <LabelTitle>{labelText}</LabelTitle>
+      </Label>
+      <BaseInput
+        className={[className, success ? 'Mui-success' : ''].join('')}
+        {...inputProps}
+      />
+      <Typography
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:81]: If you will render this block, even with text = "", marginTop will move content anyway
Снимок экрана 2022-09-02 в 13 35 52

Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx]: Done

File: src/components/Input/Input.tsx [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+import FormControl from '@mui/material/FormControl';
+import InputBase, { InputBaseProps } from '@mui/material/InputBase';
+import { styled } from '@mui/material/styles';
+
+import { ErrorIcon } from 'assets/svg';
+
+import Typography from '../Typography/Typography';
+
+type InputCustomProps = {
+  errorText?: string;
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:12]:
```suggestion
  error?: string;
```

File: src/components/Input/Input.tsx [hunk 9]
```diff
@@ -0,0 +1,96 @@
...
+  labelText?: string;
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:13]:
```suggestion
  label?: string;
```

File: src/components/Input/Input.tsx [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+};
+const Label = styled('label')(() => ({
+  color: '#6D7692',
+  lineHeight: '16px',
+  marginBottom: '5px',
+  height: 16,
+}));
+
+const LabelTitle = styled('span')(() => ({
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:22]:
```suggestion
const LabelTitle = styled(Typography)(() => ({
```

**Review by p1geon — APPROVED** [review]

File: src/components/Input/Input.tsx (hunk 10 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx]: Done

File: src/components/Input/Input.tsx (hunk 9 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx]: Done

File: src/components/Input/Input.tsx (hunk 8 above)
Comment (mr-teddy-bear) [reply src/components/Input/Input.tsx]: Done

File: src/components/Input/Input.tsx [hunk 11]
```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+
+import FormControl from '@mui/material/FormControl';
+import InputBase, { InputBaseProps } from '@mui/material/InputBase';
+import { styled } from '@mui/material/styles';
+
+import { ErrorIcon } from 'assets/svg';
+
+import Typography from '../Typography/Typography';
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:9]:
```suggestion
import Typography from 'components/Typography/Typography';
```

Comment (gavrilin-remote) [reply src/components/Input/Input.tsx:9]: suggestion

File: src/components/Input/Input.tsx [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+}: InputBaseProps & InputCustomProps): JSX.Element => {
+  return (
+    <FormControl fullWidth error={!!error} variant='standard'>
+      {!!label && (
+        <Label htmlFor={inputProps.id}>
+          <LabelTitle text={label} />
+        </Label>
+      )}
+      <BaseInput {...inputProps} />
+      {error && (
```
Comment (gavrilin-remote) [inline src/components/Input/Input.tsx:86]: suggestion
```suggestion
      {!!error && (
```

**Review by gavrilin-remote — APPROVED** [review]

**Pull request merged**